dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumActionScope.kt

```diff
@@ -82,26 +70,74 @@ internal class RumActionScope(
 
     // region Internal
 
+    private fun onViewTreeChanged(now: Long) {
+        lastInteractionNanos = now
+        viewTreeChangeCount++
+    }
+
+    private fun onStopView(
+        now: Long,
+        writer: Writer<RumEvent>
+    ) {
+        isViewStopped = true
+        ongoingResourceKeys.clear()
+        sendAction(now, writer)
+    }
+
+    private fun onStopAction(
+        event: RumRawEvent.StopAction,
+        now: Long,
+        writer: Writer<RumEvent>
+    ) {
+        name = event.name
+        attributes.putAll(event.attributes)
+        sendAction(now, writer)
+    }
+
+    private fun onStartResource(
+        now: Long,
+        event: RumRawEvent.StartResource
+    ) {
+        lastInteractionNanos = now
+        resourcesCount++
+        ongoingResourceKeys.add(WeakReference(event.key))
+    }
+
+    private fun onStopResource(
+        event: RumRawEvent.StopResource,
+        now: Long
+    ) {
+        val keyRef = ongoingResourceKeys.firstOrNull { it.get() == event.key }
+        if (keyRef != null) {
+            ongoingResourceKeys.remove(keyRef)
+            lastInteractionNanos = now
+        }
+    }
+
     private fun sendAction(
         endNanos: Long,
         writer: Writer<RumEvent>
     ) {
-        val eventData = RumEventData.UserAction(
-            name,
-            actionId,
-            max(endNanos - startedNanos, 1L)
-        )
-
-        val event = RumEvent(
-            getRumContext(),
-            eventTimestamp,
-            eventData,
-            RumFeature.userInfoProvider.getUserInfo(),
-            attributes
-        )
-
-        writer.write(event)
-        parentScope.handleEvent(RumRawEvent.SentAction(), writer)
+        if (sent) return
+
+        if (resourcesCount > 0 || viewTreeChangeCount > 0 || isViewStopped) {
```

Comment:
so you are sending the action is the View stopped. If you did not have any resources or tree changes that will mean you are sending an `empty` action and will contradict your PR. Maybe I am missing something...